### PR TITLE
Add reference token security to IP endpoints

### PR DESCRIPTION
**Pull request by chris-mannix-cko**
# Description of changes in PR

Adds 'ApiSecretKey' security to each IP endpoint, including the scope required for the NAS reference token.

## Checklist

- [x] Added a changelog entry to the `changelog.md` file in either `abc_spec` or `nas_spec`
- [x] Contacted the Tech Docs team about any corresponding guides that need to be updated for www.checkout.com/docs

**Contributing options**: Look at [our documentation](https://checkout.atlassian.net/wiki/spaces/PD/pages/2169506663/API+ref+publication+process) for options to contribute to the API reference.

### CODE REVIEW

**Comment by chris-mannix-cko** [comment]
Can be released whenever, API key support is already enabled

File: nas_spec/changelog.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Changelog
 
 | Date       | Description of change                                                                                                 
-|------------|-----------------------------------------------------------------------------------------------------------------------|
-| 2022/11/29 | Adding prism device_session_id to payment request.                                                                    |
+| ---------- |-----------------------------------------------------------------------------------------------------------------------|
+| 2022/11/30 | Add API key security to IP endpoints.
```
Comment (natasha-moore-cko) [inline nas_spec/changelog.md:5]: Any reason why we're overwriting the Prism changelog item?

Comment (chris-mannix-cko) [reply nas_spec/changelog.md:5]: It's important the secret of the `device_session_id` never gets out

**Review by natasha-moore-cko — APPROVED** [review]

**Comment by chris-mannix-cko** [comment]
> @chris-mannix-cko Any particular reason why Secret key is enabled for onboard a [sub-entity](https://api-reference.checkout.com/#operation/onboardSubEntity) but not for update the [sub-entity](https://api-reference.checkout.com/#tag/Platforms/paths/~1accounts~1entities~1%7Bid%7D/put)
> 
> The same applies for Update a sub-entity's payout schedule, that creates a lot of discrepancies on the usage of endpoints.
> 
> We are working on UPAPI to integrate API reference with SDK's

Good spot! That's a mistake - all the endpoints in the public `Platforms` API endpoints support NAS Reference token auth (I think Balances needs an update too). I will raise a PR to correct the docs. Thanks!

**Pull request merged**